### PR TITLE
search users by email

### PR DESCRIPTION
**Pull request by UnsungHero97**
This pull requests depends on https://github.com/openlattice/openlattice/pull/170

Search by email:

Screen Shot 2021-04-27 at 1 31 16 PM

Search with wildcard `*`:

Screen Shot 2021-04-27 at 1 31 28 PM

### CODE REVIEW

File: src/core/redux/selectors/selectUserSearchResults.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ */
+
+import { Map, getIn } from 'immutable';
+
+import { USERS, USER_SEARCH_RESULTS } from '../constants';
+
+export default function selectUserSearchResults() {
+
+  return (state :Map) :Map<string, Map> => getIn(state, [USERS, USER_SEARCH_RESULTS]) || Map();
+}
```
Comment (anzioka) [inline src/core/redux/selectors/selectUserSearchResults.js:12]: annotation looks a little fuzzy

File: src/core/users/sagas/searchAllUsers.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  try {
+    yield put(searchAllUsers.request(action.id, action.value));
+
+    if (action.value === '*') {
+      throw new Error('search query cannot be "*"');
+    }
+    if (!isNonEmptyString(action.value)) {
+      throw new Error('search query cannot be empty');
+    }
```
Comment (anzioka) [inline src/core/users/sagas/searchAllUsers.js:32]: no change required, but I think it feels more natural for these two statements to be reversed (First verify that the value is a non-empty string, then verify the string is not of value *)

File: src/core/redux/selectors/selectUserSearchResults.js (hunk 1 above)
Comment (UnsungHero97) [reply src/core/redux/selectors/selectUserSearchResults.js:12]: What do you mean? The return value is a `Map` where the key is `string` and the value is `Map`.

File: src/core/users/sagas/searchAllUsers.js (hunk 2 above)
Comment (UnsungHero97) [reply src/core/users/sagas/searchAllUsers.js]: Fixed.

File: src/core/redux/selectors/selectUserSearchResults.js (hunk 1 above)
Comment (anzioka) [reply src/core/redux/selectors/selectUserSearchResults.js:12]: ah ok! makes sense!

**Review by solomontang — APPROVED** [review]

**Review by bergieboy — APPROVED** [review]

**Review by giuliacampana — APPROVED** [review]

**Pull request merged**